### ISSUE TITLE
Very large stub files are not processed by Pylance.

### ISSUE THREAD

**Issue by Ryan-DowlingSoka** [microsoft/pylance-release#1520]
<h3> Environment data </h3>

-   Language Server version: 2021.6.3
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.7.7
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: off

<h3> Expected behaviour </h3>

Intellisense works even for very large stub files.

<h3> Actual behaviour </h3>

While closed, the very large stub file is not processed by Pylance, resulting in no intellisense.

<h3> Logs </h3>

Python Language Server Log
```
[Info  - 10:58:31 AM] Pylance language server 2021.6.3 (pyright e66600b2) starting
[Info  - 10:58:31 AM] Server root directory: c:\Users\ryandow\.vscode\extensions\ms-python.vscode-pylance-2021.6.3\dist
[Error - 10:58:37 AM] File length of "[REDACTED].py" is 20948900 which exceeds the maximum supported file size of 16777216
[Error - 10:58:37 AM] File length of "[REDACTED].py" is 20948900 which exceeds the maximum supported file size of 16777216
```

---

We work with a third party piece of software which creates a single stub file for all the python bindings it exposes. I've filed a bug with them that the stub file is too large, but at the same time, I am coming at it from this angle that this file limit might be an arbitrary one?

When the stub file is open in VsCode intellisense works, implying to me this is an arbitrary limit. Perhaps this could be changed in configuration?

**Comment by erictraut** [comment]
You have a stub file that is 20MB in size? A stub file is meant to describe the interface surface for a library. Is the interface to this library really that large?

We recently added this limit for practical reasons. It is somewhat arbitrary, but if we attempt to process bigger files, at some point we will hit memory or stack limits and you will experience crashes.

**Comment by Ryan-DowlingSoka** [comment]
The stub file in question is auto-generated by the software as almost a wrapper for the entire software package. The interface surface is indeed gigantic. It isn't under my domain, so unfortunately I can't re-engineer it myself to be split up, but I have put in a support ticket to have them investigate.

It also isn't written ideally (it isn't a .pyi for example, and it doesn't use typehinting explicitly like it should, which I've also requested in a ticket to them) and currently is mainly used for auto-complete, and doc strings.

But yes, the file is 416056 lines long. My hope is the vendor can refactor the code generation to use multiple files and better type-hinting, but in the meantime I don't have a ton of say on that. :P Which is why i'm coming at it from both sides here.

**Comment by erictraut** [comment]
I think we can safely increase the maximum source file size from 16MB to 32MB. I just made that change, and it will be in the next release of pylance. I don't think we should increase it any further though, so please continue to work with the vendor to produce stub files that are not so unruly.

**Comment by Ryan-DowlingSoka** [comment]
Thanks, I appreciate it. I will continue to put pressure there. 

Cheers

**Comment by jakebailey** [comment]
Leaving open until the fix is released.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.7.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202170-6-july-2021
